client/lib/jetpack/wpcom-upsell-controller.tsx

```diff
@@ -8,22 +8,43 @@ import React from 'react';
  */
 import { isEnabled } from 'config';
 import type { Context } from './types';
+import { isBusinessPlan } from 'lib/plans';
+import { isEligibleForAutomatedTransfer } from 'state/automated-transfer/selectors';
+import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
+import { getSitePlan, isJetpackSite } from 'state/sites/selectors';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { isJetpackSite } from 'state/sites/selectors';
 
-export default function upsellSwitch( UpsellComponent: typeof React.Component ): Function {
+export default function upsellSwitch(
+	UpsellComponent: typeof React.Component,
+	BusinessATComponent: typeof React.Component
+): Function {
 	return ( context: Context, next: Function ) => {
-		if ( isEnabled( 'jetpack-cloud' ) ) {
-			return next();
-		}
 		const getState = context.store.getState;
 		const siteId = getSelectedSiteId( getState() );
 		const isJetpack = isJetpackSite( getState(), siteId );
 
-		if ( ! isJetpack ) {
+		if ( isJetpack || isEnabled( 'jetpack-cloud' ) ) {
+			return next();
+		}
+
+		const currentPlan = getSitePlan( getState(), siteId );
+		const isBusiness = isBusinessPlan( currentPlan.product_slug );
+
+		// Show Business plan upsell if it isn't a Jetpack site and it doesn't have
+		// a Business plan
+		if ( ! isBusiness ) {
```

Comment:
What about eCommerce plan? That plan has 'Jetpack advanced features' like the Business plan

Reply:
Good question. I guess we still have to figure out what is the target audience for this.

Reply:
eCommerce sites are also Atomic by essence, so they'd have the same offering I believe.

Reply:
^^ @rcanepa I'd add it now and it'd be good to have a specific task (testing task) to walk through all these cases together with a "product owner" aka, Luca or Paolo.
@keoshi

Reply:
Agreed with what Clemente said above! 🙌

Reply:
Yes. Since eCommerce sites are AT sites, they are Jetpack sites as well. This means they will comply with `if ( isJetpack || isEnabled( 'jetpack-cloud' ) )`, and their users will see the Backup or Scan page directly (without going through the upsell or Automated Transfer page).

Reply:
If the site is not a Jetpack site (self-hosted Jetpack, Business Atomic, and eCommerce sites are included here), we check if it is a Business site. If it is true, then we offer a path to the Automated Transfer process. If not, it means the site has a Free, Personal or Premium plan, therefore, we offer an upgrade to Business.

Reply:
mmmm Yes, I've checked it now with a site with eCommerce plan, the Backup/Scan section is enabled but I get this upsell page:

![image](https://user-images.githubusercontent.com/9832440/84886857-e0b3a100-b08c-11ea-80fb-dc1c21e17333.png)

And the Scan section doesn't load...

I didn't trigger any auto-transfer process (there isn't upsell page to do it). 

@keoshi is this expected?, if not, maybe we could add this case to ask to someone.

Reply:
Not expected @cleacos, but to be fair I got the same page when AT'ing from a site with a Business plan to the Atomic platform.

> I didn't trigger any auto-transfer process (there isn't upsell page to do it).

Note that eCommerce sites are Atomic by definition since they already have WooCommerce installed.

I think this situation has more to do with the AT process than anything else, but I could be wrong of course. 😬

Reply:
Yes, I checked it later and I realized that eCommerce plan is an AT site. It needs more investigation

Reply:
Yes. It seems that the AT process takes some time. I wrote this `pbtFFM-jq-p2` to surface the problem and see what can be done this fix this intermediate state.